src/index.js

```diff
@@ -230,7 +236,7 @@ export default class VueI18n {
       if (isPlainObject(message)) {
         ret = message[key]
         if (typeof ret !== 'string') {
-          if (process.env.NODE_ENV !== 'production' && !this._silentTranslationWarn) {
+          if (process.env.NODE_ENV !== 'production' && !this._silentTranslationWarn && !(this._silentFallbackWarn && (this._isFallbackRoot() || locale !== this.fallbackLocale))) {
```

Comment:
> is it ok to check `locale !== this.fallbackLocale`? 

Yes, that's right!

>I got confused because `_translate` gets `fallback` as a param, but it always points to `this.fallbackLocale`.

The current vue-i18n is very complexity due to none of your business. 😭 In next major version, I will to write with full-scratch.

Reply:
> Also, we have a similar condition in line 246/252.

Just as you said, these is similar condition.
In unit test coverage (the following is  removed `/* istanbul ignore else */`), the code of 246/249 is not passed in unit tests. 

![image](https://user-images.githubusercontent.com/72989/51516199-df5f5400-1e59-11e9-86cd-8108143de1a2.png)


However, `t` or `$t` is returned the value of `TranslateResult`

https://github.com/kazupon/vue-i18n/blob/e8790244408d812728c95253b63b637a734d1c32/types/index.d.ts#L12

If you want to add a test case, we will be very helpful if you do so.
Although it is a bad design as API 😂, We need to maintain backwards compatibility.

In next major version, I'll plan to change this specification.